src/isar/state_machine/states/stop_step.py

```diff
@@ -43,17 +48,32 @@ def _run(self):
             try:
                 self.stop_thread.get_output()
             except ThreadedRequestNotFinishedError:
-                time.sleep(self.state_machine.sleep_time)
-                continue
+                if self._has_failed_current:
+                    transition = self.state_machine.mission_stopped
+                    self.logger.warning(
+                        "Could not communicate request: Reached limit for stop attemps. Cancelled mission and transitioned to idle."
+                    )
+                    break
+                else:
+                    self.logger.info("Attempting to stop mission.")
+                    self.handle_stop_fail(retry_limit=50)
```

Comment:
Could this function return the _has_failed_current to be used straight in the if statement at line 51?